openstudiocore/CMakeLists.txt

```diff
@@ -551,46 +561,32 @@ if(BUILD_WITH_OPENSSL)
   endif()
 endif()
 
-# swig
-
-set(SWIG_INSTALL_DIR "" CACHE PATH "Path to swig Install")
-mark_as_advanced(SWIG_INSTALL_DIR)
-
-if(NOT EXISTS ${SWIG_INSTALL_DIR})
-  if( WIN32 )
-    set(SWIG_ZIP_FILENAME "swig_3_0_10_Win32.zip")
-    set(SWIG_ZIP_EXPECTED_MD5 "be3c42e6daffde06a05ad1b978338ebd")
-  elseif( APPLE )
-    set(SWIG_ZIP_FILENAME "swig_3_0_10_osx.tar.gz")
-    set(SWIG_ZIP_EXPECTED_MD5 "f0ea8749f8a0345ee37e1766d27a984d")
-  elseif(EXISTS "/etc/redhat-release")
-  else()
-  endif()
-
-  set(SWIG_ZIP_LOCAL_PATH "${CMAKE_BINARY_DIR}/${SWIG_ZIP_FILENAME}")
-  if(EXISTS "${SWIG_ZIP_LOCAL_PATH}")
-    file(MD5 "${SWIG_ZIP_LOCAL_PATH}" SWIG_ZIP_MD5)
-  endif()
+# SWIG
 
-  if( NOT SWIG_ZIP_MD5 STREQUAL SWIG_ZIP_EXPECTED_MD5)
-    file(DOWNLOAD "http://openstudio-resources.s3.amazonaws.com/dependencies/${SWIG_ZIP_FILENAME}"
-      ${SWIG_ZIP_LOCAL_PATH}
-      INACTIVITY_TIMEOUT 120
-      SHOW_PROGRESS
-      EXPECTED_MD5 ${SWIG_ZIP_EXPECTED_MD5})
-      execute_process(COMMAND ${CMAKE_COMMAND} -E tar xfz ${SWIG_ZIP_LOCAL_PATH} WORKING_DIRECTORY "${CMAKE_BINARY_DIR}")
-  endif()
-
-  set(SWIG_INSTALL_DIR "${CMAKE_BINARY_DIR}/Swig-install/")
+if(UNIX)
+  # We patch it up with a version of pcre we provide to avoid having to have the requirement locally
+  ExternalProject_Add(SWIG
```

Comment:
This was the quickest way to move it into core since 1) binary packages weren't portable and 2) SWIG is a small dependency not worth messing around maintaining at the super level.  

I guess the thing to do on Unix is to download the source tar at configure time, then patch and build SWIG as a custom command with output being swig.exe, then you would still need to make those other commands depend on swig.exe.  You could so something similar on Windows but download the binary instead of source.  To me external project does all these steps in an easy way, only real difference is it downloads at build time instead of configure, right?

Other thing to look into is if we even need a custom SWIG build anymore, if Jason's commits were merged into SWIG maybe we can just use the normal releases?